### PR TITLE
plasma cutter rework 1

### PR DESCRIPTION
**Pull request by aceedex**
[content] [tweak]

This PR changes all instances of our current plasma cutter to a plasma torch. functionally identical but with a new name and altered description. What this also adds is a child of that item called the plasma cutter. This new plasma cutter combines the features of a diamond mining drill and a proto kinetic accelerator. To assemble it, you'll need a plasma cutter conversion kit printable from research. The conversion kit has the same requirements as our current plasma cutter except that it also needs NT research 1. 

The modkit however starts without parts in it, youll need to use a proto kinetic accelerator on it, then use the modkit on a diamond mining drill to receive a plasma cutter.

Compared to a proto kinetic accelerator, aside from the diamond drill functionality, the plasma cutter has a higher shot count. 10 on a PKA to the cutter's 15. The plasma cutter is also not rechargeable anymore. instead it can be loaded by adding plasma sheets into it. One shot per sheet. 

BIG UPDATE:
I do agree that NT tech is a bit much for something that's just in practice, more PKAs. 
Plasma cutters now also use a new projectile. Identical to PKAs but deals 15 extra damage to simple mobs and xenomorphs.

:cl:
 * rscadd:  A new version of the plasma cutter has been added. You'll need a modkit from research to make it
 * tweak: The old plasma cutter has been renamed to a plasma torch

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
I'm boy2mantwicethefam, and this is my favorite pull request on /vg/.

**Comment by codeungabunga** [comment]
You didn't update the NT disk crate research paper which mentions all the techs you can make with it.

Why all these hoops to jump through just to get a ranged weapon when you can just ask science at that point to print a plasma/laser cannon/rifle?

**Comment by aceedex** [comment]
because.
1. roid mobs are resistant to RD weapons
2. it encourages types of research that arent seen often. plasma and NT
3. mining can get the NT disk roundstart. With that in mind, I dont believe it be so crazy a requirement. But if you feel this is anomaly research tier, Im more than happy to compromise on the requirement
4. it keeps mining progression intact. cant skip the 1000s of dollars needed for a pka by just printing these

keep in mind, a stack of plasma and a plasma cutter take the place of 6 proto kinetic accelerators in shot count

good catch on the paper, Ill get on it

**Comment by SNilWillIllShill** [comment]
>making new plasma cutter the Dead Space plasma cutter
FUCK YOU AND FUCK YOUR MARKER
good PR

**Comment by SonixApache** [comment]
add a new projectile to it that has sharpness or something so it could actually cut limbs off xenos

**Comment by aceedex** [comment]
I do have plants for a variant that deals bonus daage to limbs, but first Id like for this to get ironed out in case I need to make more changes

**Comment by aceedex** [comment]
new changes, see OP

File: code/game/objects/items/devices/modkit.dm [hunk 1]
```diff
@@ -147,3 +147,22 @@
...
 	original[1] = /obj/item/weapon/gun/energy/gun
 	finished[1] = /obj/item/weapon/gun/energy/gun/nuclear
+
+/obj/item/device/modkit/plasmacutter
+	name = "plasma cutter conversion kit"
+	desc = "A set of tools that enables conversion of a mining diamond drill into a plasma cutter. Needs to be loaded with the parts of a proto-kinetic accelerator first."
+
+/obj/item/device/modkit/plasmacutter/New()
+	..()
+	parts = new/list(1)
```
Comment (ShiftyRail) [inline code/game/objects/items/devices/modkit.dm:157]: I can't really ask you to change the whole file for your update, but I'm not really sure why those are lists

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -263,11 +263,60 @@ proc/move_mining_shuttle()
...
+	if(proximity_flag && istype(target, /obj/item/stack/sheet))
+		var/obj/item/stack/sheet/A = target
+		if(A.mat_type == MAT_PLASMA)
+			if(current_ammo < max_ammo)
+				var/loading_ammo = max_ammo - current_ammo
+				A.amount = A.amount - loading_ammo
+				current_ammo = current_ammo + loading_ammo
+				to_chat(user, "You load [src].")
+			else
+				to_chat(user, "[src] is already loaded")
```
Comment (ShiftyRail) [inline code/modules/mining/mine_items.dm]: Missing a  dot.
Could also wrap those in `<span class='notice'>`.

File: code/modules/mining/mine_items.dm [hunk 3]
```diff
@@ -263,11 +263,60 @@ proc/move_mining_shuttle()
...
+		src.visible_message("*click click*")
+		playsound(src, 'sound/weapons/empty.ogg', 100, 1)
+
+/obj/item/weapon/pickaxe/plasmacutter/accelerator/attackby(atom/target, mob/user, proximity_flag)
+	if(proximity_flag && istype(target, /obj/item/stack/sheet))
+		var/obj/item/stack/sheet/A = target
+		if(A.mat_type == MAT_PLASMA)
+			if(current_ammo < max_ammo)
+				var/loading_ammo = max_ammo - current_ammo
+				A.amount = A.amount - loading_ammo
```
Comment (ShiftyRail) [inline code/modules/mining/mine_items.dm]: `A.amount -= loading_ammo`  reads better, in my opinion.

File: code/modules/projectiles/projectile/special.dm [hunk 4]
```diff
@@ -278,6 +278,13 @@ obj/item/projectile/kinetic/New()
 	spawn(4)
 		returnToPool(src)
 
+/obj/item/projectile/kinetic/cutter
+
+/obj/item/projectile/kinetic/cutter/to_bump(atom/A as mob|obj|turf|area)
```
Comment (ShiftyRail) [inline code/modules/projectiles/projectile/special.dm]: `as ...` is unnecessary, you can remove it.

File: code/game/objects/items/devices/modkit.dm (hunk 1 above)
Comment (aceedex) [reply code/game/objects/items/devices/modkit.dm]: I think deity made those so you can ask him. My guess: ability to make a modkit have different types of parts for different modifications

**Comment by DeityLink** [comment]
I like giving scientists more reason to pay for Nanotrasen Tech. I feel like many people don't even know this exists.

File: code/game/objects/items/devices/modkit.dm (hunk 1 above)
Comment (Exxion) [reply code/game/objects/items/devices/modkit.dm:157]: This is also creating a list with a first element of 1, not a list of length 1, but whatever

File: code/modules/mining/mine_items.dm [hunk 5]
```diff
@@ -263,11 +263,60 @@ proc/move_mining_shuttle()
...
+
+/obj/item/weapon/pickaxe/plasmacutter/accelerator/attackby(atom/target, mob/user, proximity_flag)
+	if(proximity_flag && istype(target, /obj/item/stack/sheet))
+		var/obj/item/stack/sheet/A = target
+		if(A.mat_type == MAT_PLASMA)
+			if(current_ammo < max_ammo)
+				var/loading_ammo = max_ammo - current_ammo
+				A.amount -= loading_ammo
+				current_ammo += loading_ammo
+				to_chat(user, "<spanclass='notice'>You load [src].</span>")
```
Comment (ShiftyRail) [inline code/modules/mining/mine_items.dm]: `<span class='...'>`, not `<spanclass ...>`, please

Comment (aceedex) [reply code/modules/mining/mine_items.dm]: @ShiftyRail done. worked as it was when I tested though I swear on me mum

**Comment by DamianX** [comment]
what the fuck happened here

**Comment by ShiftyRail** [comment]
What do you mean

**Comment by DamianX** [comment]
the `parts`, `original`, `finished` are initialized in the weirdest way possible
`attackby()` doesn't even take `proximity_flag` as an argument anywhere

**Pull request merged**